3.0 iproto uri params (#3973)
tarantool/doc: doc/reference/configuration/configuration_reference.rst

.. confval:: iproto_advertise.<peer_or_sharding>.uri

>     (Optional) An URI used to advertise the current instance.

Reviewer: ```suggestion
    (Optional) A URI used to advertise the current instance.
```

Author: Thanks, also fixed in other places

.. confval:: iproto_advertise.<peer_or_sharding>.login

>     (Optional) A username that should be used to connect to the current instance.

Reviewer: ```suggestion
    (Optional) A username used for connecting to the current instance.
```
Feel free to ignore this edit if it changes the meaning.

Author: Changed to `A username used to connect to the current instance.`

.. confval:: iproto_advertise.<peer_or_sharding>.password

    (Optional) A password for the specified user.

>     If ``login`` is specified but a password is missing, it is taken from :ref:`credentials <configuration_reference_credentials>` for this user.

Reviewer: ```suggestion
    If a ``login`` is specified but a password is missing, it is taken from the user's :ref:`credentials <configuration_reference_credentials>`.
```

        -   Another Tarantool instance from this cluster. This means that one instance might act as the server that accepts connections from other instances and the client that connects to other instances.
        -   A remote administrative console (:ref:`tt connect <tt-connect>`).
        -   A :ref:`net.box <net_box-module>` connector.
        -   :ref:`Connectors <index-box_connectors>` provided for different languages.

    ``<uri>.params.transport`` accepts one of the following values:

    -   ``plain`` (default): turn off traffic encryption.
    -   ``ssl``: encrypt traffic by using the TLS 1.2 protocol (`Enterprise Edition <https://www.tarantool.io/compare/>`_ only).

    **Example**

    The example below demonstrates how to enable traffic encryption for connections between replica set peers.
    The following parameters are specified for each instance:

    -   ``ssl_ca_file``: a path to a trusted certificate authorities (CA) file.
    -   ``ssl_cert_file``: a path to an SSL certificate file.
    -   ``ssl_key_file``: a path to a private SSL key file.
    -   ``ssl_password`` (``instance001``): a password for an encrypted private SSL key.

>         For ``instance002`` and ``instance003``, passwords for encrypted SSL keys are provided in ``ssl_password_file``.

Reviewer: Maybe make `ssl_password_file` a separate list item for better visibility?
